examples/js/modifiers/BufferSubdivisionModifier.js

```diff
@@ -0,0 +1,753 @@
+/*
+ *	@author zz85 / http://twitter.com/blurspline / http://www.lab4games.net/zz85/blog 
+ *	@author Matthew Adams / http://www.centerionware.com - added UV support and rewrote to use buffergeometry.
+ *
+ *	Subdivision Geometry Modifier 
+ *		using Loop Subdivision Scheme for Geometry / BufferGeometry
+ *
+ *	References:
+ *		http://graphics.stanford.edu/~mdfisher/subdivision.html
+ *		http://www.holmes3d.net/graphics/subdivision/
+ *		http://www.cs.rutgers.edu/~decarlo/readings/subdiv-sg00c.pdf
+ *
+ *	Known Issues:
+ *		- currently doesn't handle "Sharp Edges"
+ *      - no checks to prevent breaking when uv's don't exist.
+ *      - vertex colors are unsupported.
+ *     **DDS Images when using corrected uv's passed to subdivision modifier will have their uv's flipy'd within the correct uv set
+ *     **Either flipy the DDS image, or use shaders. Don't try correcting the uv's before passing into subdiv (eg: v=1-v).
+ *
+ * @input THREE.Geometry, or index'd THREE.BufferGeometry with faceUV's (Not vertex uv's)
+ * @output non-indexed vertex points, uv's, normals.
+ *     
+ */
+
+/*
+ *
+ * The TypedArrayHelper class is designed to assist managing typed arrays, and to allow the removal of all 'new Vector3, new Face3, new Vector2'.
+ * 
+ * It will automatically resize them if trying to push a new element to an array that isn't long enough
+ * It provides 'registers' that the units can be mapped to. This allows a small set of objects
+ * (ex: vector3's, face3's, vector2's) to be allocated then used, to eliminate any need to rewrite all
+ * the features those classes offer while not requiring some_huge_number to be allocated.
+ * It should be moved into it's own file honestly, then included before the BufferSubdivisionModifier - maybe in three's core?
+ *
+ *
+ * EX: new TypedArrayHelper(initial_size_in_elements, 3, THREE.Vector3, Float32Array, 3, ['x', 'y', 'z']); (the x,y,z comes from THREE.Vector3. It would be abc if it were a face3. etc etc)
+ *
+ */
+THREE.Face3.prototype.set = function (a, b, c) {
+    this.a = a;
+    this.b = b;
+    this.c = c;
+}
+var TypedArrayHelper = function (size, registers, register_type, array_type, unit_size, accessors) {
+    this.array_type = array_type;
+    this.register_type = register_type;
+    this.unit_size = unit_size;
```

Comment:
But that may not always be ideal so I will leave it. Eg: when using vec4 registers on a vec3 typed array for some reason I can't think of right now.